mathesar/api/db/viewsets/records.py

```diff
@@ -199,6 +199,8 @@ def partial_update(self, request, pk=None, table_pk=None):
 
     def destroy(self, request, pk=None, table_pk=None):
         table = get_table_or_404(table_pk)
+        if table.get_record(pk) is None:
+            raise NotFound
```

Comment:
`404` could be due to a missing Table or it could be due to a missing record. Even though the `NotFound` exception class could work in this context because the frontend does not care about the object that is missing right now, it would be better to use `NotFoundAPIException` as it would enable the backend to send a more useful error through the `error_code` property of the `NotFoundAPIException` on top of the `404 status code`

On a side note, `get_table_or_404` should be using `NotFoundAPIException` but it is still using `NotFound` because it was created before the exception handler was created and refactoring it is not a priority atm

Reply:
Ah, I did raise the `NotFoundAPIException` before(in my previous commit). I'll revert it.

Reply:
I also found out `NotFound` getting raised here: https://github.com/centerofci/mathesar/blob/091ae33d4e354d80e9872f06d77649b548a3acb9/mathesar/api/db/viewsets/records.py#L123-L124

and 

https://github.com/centerofci/mathesar/blob/a45c557085c609aef12eaf38cc9601d8153540a6/mathesar/api/db/viewsets/columns.py#L172-L173